src/client.ts

```diff
@@ -1816,7 +1840,12 @@ export class Client<Ctx = null> {
       message: 'status:reconnecting',
     });
 
-    this.connect({ tryCount: 0, websocketFailureCount: 0 });
+    this.connectionId++; // TODO: should will reconnects increment this or not?
```

Comment:
should will!?

Reply:
does is have will???

Reply:
I can't ship till I answer the question I asked there. On it.

Reply:
oh it isn't as incoherent as I thought either. That's "should `willReconnect`s increment this "

Reply:
At this point, we've unambiguously cleaned up the handler, so incrementing is correct.

```suggestion
    this.connectionId++;
```